### DISCUSSION TITLE
Spice client folder never becomes available

### DISCUSSION THREAD

**Discussion by brannerchinese** [utmapp/UTM#3917] [UTM Help (macOS)]
I am having trouble accessing the shared folder from Ubuntu Desktop. When I open the `File` application and choose "Other locations", under "Networks" I see an item "Spice client folder". However, no matter how long I wait, any efforts to open it bring "Unable to access location" and "HTTP error: Could not connect: Connection refused". 

Is there a URL or path of some sort for the shared folder, once it has become available? 

Is there something further I need to do? Is there some sort of manual device-mounting process that also needs to take place?

Configuration:

* Host: MacOS Monterey 12.3.1
* Guest: Ubuntu 20.04 Focal (desktop)
* `spice-vdagent` `spice-webdavd` are installed and running
* Under UTM's settings for this Guest, under "Sharing", both "Enable Clipboard Sharing" and "Shared Directory" are checked.

**Comment by lonewolf2222** [comment]
I have the same problem but with Kali Linux as guest OS. Clicking on "Spice client folder" does not open anything. 

Host: MacOS 12.3.1
Host: Kali Linux 2022.1 
Networking: Bridged

Networking is working. In the guest OS, I can also access AFP shares from my NAS located on my local network.

**Reply by pawelsulik** [reply]
I fixed by 

apt-get install spice-*

**Reply by varunpan** [reply]
How can i access the shared directory from CLI?

**Reply by s3784244** [reply]
I'm having the same issue, I'm not able to access http://127.0.0.1:9843/. Any other solutions?

**Comment by MarcoBonino** [comment]
Same problem.
I tried `apt-get install spice-*` and to uncheck/check `Enable Clipboard Sharing` and `Shared Directory`.

- Host: MacOS Monterey 12.4 (M1 chip)
- Guest: Ubuntu Server 22.04 AArch64

**Comment by rcrx** [comment]
Is there a resolution to this issue?

**Comment by simonhuwiler** [comment]
Same problem here. At first, it worked. After a reboot: "Connection refused". I can see the folder via browser: http://127.0.0.1:9843/

Host: MacOS BigSur
VM: Ubuntu 20.04

**Comment by thisisrandy** [comment]
All of the proposed fixes floated here have "worked"  for me. However, none of them are reliable. Ultimately, the root cause seems to be time. If I wait an unspecified amount of time before accessing the Spice client folder,  it works.  Otherwise, it issues the error OP specifies. There is clearly a bug here that needs to be addressed, but for anyone coming here for an immediate fix, wait a while, and the Spice client folder will resolve. Unfortunately, I cannot offer anything more specific at this time.

EDIT: While this works, @onedrawingperday's [solution](https://github.com/utmapp/UTM/discussions/3917#discussioncomment-4081185) is much more direct. Newcomers to this issue, please see that reply.

I'll also note that @onedrawingperday's solution is bufferless, which is a huge deal when transferring large files on a low memory host machine. The Spice folder, by contrast, fills a several GB buffer which can easily exhaust memory.

**Comment by oliviajohansen** [comment]
All the above didn't really work for me but this did: follow the below step from https://mac.getutm.app/gallery/ubuntu-20-04

<img width="1173" alt="Screenshot 2022-09-02 at 1 01 00 AM" src="https://user-images.githubusercontent.com/55739990/187971117-f83b97a2-3724-4ddd-84a8-51b5edc46af8.png">

then run 
`sudo netplan generate`
`sudo netplan apply`

and test that the networking works using
`ping -c3 www.ubuntu.com`

**Comment by onedrawingperday** [comment]
For the past year and a half,  I've been using UTM on a Mac host with a Ubuntu LTS guest and never encountered this issue with the Spice client folder, until today, right after I upgraded to Mac OS Ventura.

After trying out various workarounds, I stumbled upon [this comment](https://gitlab.gnome.org/GNOME/gnome-boxes/-/issues/430#note_646756) over at the Gnome forums and managed to solve the issue by typing `dav://localhost:9843` in Nautilus `Other Locations > Connect to Server`. Also I bookmarked the mounted directory in the Nautilus sidebar, so that I don't have to enter the above again.

This looks like an odd Ubuntu/Gnome bug.

**Reply by stuaxo** [reply]
I had to do the same with a debian 32 bit guest in an ubuntu 64 bit host.

The `dav://localhost:9843` goes in nautilus in the guest machine.

**Comment by fiore42** [comment]
None of these suggestions worked for me.

I finally found joy following [these steps](https://docs.getutm.app/guest-support/linux/#virtfs).

Host: MacOS Ventura 13.0 on MacBook Air M2
Guest: Ubuntu 22.04 for ARM
UTM 4.0.9

hope it helps!

**Reply by martin-braun** [reply]
`curl http://127.0.0.1:9843` returns my files, so far so good, but how do I mount properly with `davfs2`? It asks for credentials which I don't have, I tried my credentials and the credentials of the host machine.

EDIT: Ok it works, but it took very long to authenticate.

**Comment by pakyrs** [comment]
works fine, use these steps:

```
# install the spice-webdav tools
apt install spice-webdavd

# check if the folder share works
curl http://127.0.0.1:9843

# install davfs2
apt install davfs2

# create a folder to mount to
mkdir -p /mnt/webdav

# mount the webdav share, use your client username / password to login
mount.davfs http://127.0.0.1:9843 /mnt/webdav
```

For better performance and easy management use VirtFS on linux hosts.
